### PR TITLE
test(codecv1): add zkEVM standard challenge digest test vectors

### PR DESCRIPTION
**Pull request by Thegaram**
### Purpose or design rationale of this PR

Add some test vectors from the Rust implementation of codecv1.

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [X] test: Adding missing tests or correcting existing tests

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [X] No, this PR doesn't involve a new deployment, git tag, docker image tag
- [ ] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [X] No, this PR is not a breaking change
- [ ] Yes

### CODE REVIEW

**Review by roynalnaruto — DISMISSED** [review]

**Review by colinlyguo — DISMISSED** [review]
LGTM. Are the assertions used for future changes not breaking these assertions?

File: common/types/encoding/codecv1/codecv1_test.go [hunk 1]
```diff
@@ -528,6 +529,69 @@ func TestCodecV1BatchChallenge(t *testing.T) {
...
+		// multiple non-empty chunks
+		{chunks: [][]string{{"0x010203"}, {"0x070809"}}, expected: "60376321eea0886c29bd97d95851c7b5fbdb064c8adfdadd7678617b32b3ebf2"},
+		// empty chunk followed by non-empty chunk
+		{chunks: [][]string{{}, {"0x010203"}}, expected: "054539f03564eda9462d582703cde0788e4e27c311582ddfb19835358273a7ca"},
+		// non-empty chunk followed by empty chunk
+		{chunks: [][]string{{"0x070809"}, {}}, expected: "0b82dceaa6ca4b5d704590c921accfd991b56b5ad0212e6a4e63e54915a2053b"},
+		// max number of chunks all empty
+		{chunks: [][]string{{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}}, expected: "174cd3ba9b2ae8ab789ec0b5b8e0b27ee122256ec1756c383dbf2b5b96903f1b"},
+		// max number of chunks all non-empty
+		{chunks: [][]string{{}, {"0x0a"}, {"0x0a0b"}, {"0x0a0b0c"}, {"0x0a0b0c0d"}, {"0x0a0b0c0d0e"}, {"0x0a0b0c0d0e0f"}, {"0x0a0b0c0d0e0f10"}, {"0x0a0b0c0d0e0f1011"}, {"0x0a0b0c0d0e0f101112"}, {"0x0a0b0c0d0e0f10111213"}, {"0x0a0b0c0d0e0f1011121314"}, {"0x0a0b0c0d0e0f101112131415"}, {"0x0a0b0c0d0e0f10111213141516"}, {"0x0a0b0c0d0e0f1011121314151617"}}, expected: "31d0b297e3aa9c5256afc52123ef3c1db0e9ce44a3ede66343705c86e3495d8a"},
```
Comment (colinlyguo) [inline common/types/encoding/codecv1/codecv1_test.go]: this is not an all-non-empty test case. the first chunk is empty.

File: common/types/encoding/codecv1/codecv1_test.go [hunk 2]
```diff
@@ -528,6 +529,69 @@ func TestCodecV1BatchChallenge(t *testing.T) {
...
+		// single chunk blob full
+		{chunks: [][]string{{repeat(123, nRowsData)}}, expected: "61405cb0b114dfb4d611be84bedba0fcd2e55615e193e424f1cc7b1af0df3d31"},
+		// multiple chunks blob full
+		{chunks: [][]string{{repeat(123, 1111)}, {repeat(231, nRowsData-1111)}}, expected: "22533c3ea99536b4b83a89835aa91e6f0d2fc3866c201e18d7ca4b3af92fad61"},
+		// max number of chunks only last one non-empty not full blob
+		{chunks: [][]string{{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {repeat(132, nRowsData-1111)}}, expected: "0e6525c0dd261e8f62342b1139062bb23bc2b8b460163364598fb29e82a4eed5"},
+		// max number of chunks only last one non-empty full blob
+		{chunks: [][]string{{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {repeat(132, nRowsData-1)}}, expected: "16e9355a968787a21290b0298ecf9681776d6fa97aa68c8c509b26ad301b2216"},
```
Comment (colinlyguo) [inline common/types/encoding/codecv1/codecv1_test.go]:
```suggestion
		{chunks: [][]string{{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {repeat(132, nRowsData)}}, expected: "16e9355a968787a21290b0298ecf9681776d6fa97aa68c8c509b26ad301b2216"},
```

Comment (colinlyguo) [reply common/types/encoding/codecv1/codecv1_test.go]: why minus 1 here?

File: common/types/encoding/codecv1/codecv1_test.go (hunk 1 above)
Comment (Thegaram) [reply common/types/encoding/codecv1/codecv1_test.go]: True. I copied the labels from the zkEVM test vectors.

File: common/types/encoding/codecv1/codecv1_test.go (hunk 2 above)
Comment (Thegaram) [reply common/types/encoding/codecv1/codecv1_test.go]: Not sure, I just copied the test cases, maybe @roynalnaruto knows

File: common/types/encoding/codecv1/codecv1_test.go (hunk 1 above)
Comment (roynalnaruto) [reply common/types/encoding/codecv1/codecv1_test.go]: Its a mistake. I will give you the new test vector and expected challenge

File: common/types/encoding/codecv1/codecv1_test.go (hunk 2 above)
Comment (roynalnaruto) [reply common/types/encoding/codecv1/codecv1_test.go]: Its a typo. Should be removed. I will give you the new expected challenge.

File: common/types/encoding/codecv1/codecv1_test.go (hunk 1 above)
Comment (roynalnaruto) [inline common/types/encoding/codecv1/codecv1_test.go]:
```suggestion
		{chunks: [][]string{{"0x0a"}, {"0x0a0b"}, {"0x0a0b0c"}, {"0x0a0b0c0d"}, {"0x0a0b0c0d0e"}, {"0x0a0b0c0d0e0f"}, {"0x0a0b0c0d0e0f10"}, {"0x0a0b0c0d0e0f1011"}, {"0x0a0b0c0d0e0f101112"}, {"0x0a0b0c0d0e0f10111213"}, {"0x0a0b0c0d0e0f1011121314"}, {"0x0a0b0c0d0e0f101112131415"}, {"0x0a0b0c0d0e0f10111213141516"}, {"0x0a0b0c0d0e0f1011121314151617"}, {"0x0a0b0c0d0e0f101112131415161718"}}, expected: "1e93e961cdfb4bd26a5be48f23af4f1aa8c6bebe57a089d3250f8afb1e988bf8"},
```

File: common/types/encoding/codecv1/codecv1_test.go (hunk 2 above)
Comment (roynalnaruto) [inline common/types/encoding/codecv1/codecv1_test.go]:
```suggestion
		{chunks: [][]string{{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {repeat(132, nRowsData)}}, expected: "3a638eac98f22f817b84e3d81ccaa3de080f83dc80a5823a3f19320ef3cb6fc8"},
```

**Review by roynalnaruto — COMMENTED** [review]
I've added the expected diff. Could you commit and re-run the CI?

**Review by roynalnaruto — APPROVED** [review]

**Pull request merged**